File: redact-dropwizard-example/src/main/java/pii/PiiResource.java
package pii;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.GET;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("pii")
public class PiiResource {
    private static final Logger logger = LoggerFactory.getLogger(PiiResource.class);

    public PiiResource() {
        logger.info("ctor");
    }

    @Path("{id}")
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Password getMemberInfo(@PathParam("id") Password password) {
        logger.info("password: {}", password);
        return password;
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON)
    public PiiBody put(PiiBody piiBody) {
        logger.info("piiBody: {}", piiBody);
        piiBody.redact = piiBody.toString();
        return piiBody;
    }
}

File: redact-dropwizard-kotlin-example/src/main/java/us/jsy/redact/pii/PiiResource.kt
package us.jsy.redact.pii

import org.slf4j.LoggerFactory
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("pii")
class PiiResource {
    @Path("{id}")
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun getMemberInfo(@PathParam("id") password: Password): Password {
        logger.info("password: {}", password)
        return password
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON)
    fun put(piiBody: PiiBody): PiiBody {
        logger.info("piiBody: {}", piiBody)
        piiBody.redact = piiBody.toString()
        return piiBody
    }

    companion object {
        private val logger = LoggerFactory.getLogger(PiiResource::class.java)
    }

    init {
        logger.info("ctor")
    }
}